provider/pkg/provider/resource/provider.go

```diff
@@ -12,27 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package register
+package resource
 
 import (
 	"github.com/pulumi/pulumi-kubernetes/provider/v4/pkg/clients"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 	pulumiprovider "github.com/pulumi/pulumi/sdk/v3/go/pulumi/provider"
 )
 
-type ResourceProvider interface {
+type ResourceProvider interface { // nolint:golint
 	Construct(ctx *pulumi.Context, typ, name string,
 		inputs pulumiprovider.ConstructInputs, options pulumi.ResourceOption) (*pulumiprovider.ConstructResult, error)
 }
 
-type ResourceProviderOptions struct {
+type ResourceProviderOptions struct { // nolint:golint
 	ClientSet        *clients.DynamicClientSet
 	DefaultNamespace string
 }
 
-type ResourceProviderFactory func(*ResourceProviderOptions) ResourceProvider
+type ResourceProviderFactory func(*ResourceProviderOptions) ResourceProvider // nolint:golint
 
-type ResourceProviderFuncs struct {
+type ResourceProviderFuncs struct { // nolint:golint
```

Comment:
Why do we need to disable lint here? You can use a format like below to make exceptions like this more self-explanatory:
```
//nolint:golint // Explanation.
```

(Note no space after the slashes since it's a machine-readable directive. `nolintlint` is helpful for that one.)

Reply:
Good idea, in this case for stuttering. I'll add commentary in future.